build-logic/publication/src/main/kotlin/com/avito/android/publish/CreateStagingRepositoryTask.kt

```diff
@@ -0,0 +1,105 @@
+package com.avito.android.publish
+
+import com.github.salomonbrys.kotson.fromJson
+import com.google.gson.Gson
+import com.google.gson.JsonParseException
+import okhttp3.Authenticator
+import okhttp3.Credentials
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+abstract class CreateStagingRepositoryTask : DefaultTask() {
```

Comment:
How can we test and debug this logic?

Reply:
it can be tested right in `publication/src/main/test`